### DISCUSSION TITLE
suggest page.extract_words() word sequence same as page.extract_text()

### DISCUSSION THREAD

**Discussion by rahxphoon** [jsvine/pdfplumber#1068] [Q&A]
I would like to use page.extract_words() **words' position** feature, found the extracted text sequence is not like page.extract_text()
since page.extract_text() is extracted per same line
how can I make page.extract_text() sequence same as  page.extract_text()

**Comment by jsvine** [comment]
Hi @rahxphoon, and thanks for your interest in this library. Unfortunately, I'm having some difficulty understanding the question. Could you provide additional explanation and a specific example?

**Reply by rahxphoon** [reply]
I am so sorry that I missed your msg before
take this pdf piece for example 
![image](https://github.com/jsvine/pdfplumber/assets/16117379/57d444c2-f9d2-4e5b-932b-ffa63c7c07e3)
using page.extract_text(), the extracted work would be like:
**2 DEMURRAGE - TRANSHIPMENT 1 DAY 100.00 $ 100.00 Z**
![image](https://github.com/jsvine/pdfplumber/assets/16117379/8428149d-f4aa-4cd1-8536-7133ebb6f596)

when page.extract_words() 
the sequence is like 
2 DEMURRAGE - TRANSHIPMENT 1 100.00 DAY 100.00 Z
![image](https://github.com/jsvine/pdfplumber/assets/16117379/dc5ba42b-6f30-440c-9cf3-1e387a4d320f)
**DAY** is not after **1**  is because **DAY** is not the same line(Y position) as **1**

**Reply by rahxphoon** [reply]
I would like the page.extract_words() sequence same as  page.extract_text() since I need apply some regex to extract info from the PDF, and I think page.extract_text() is more robust , since current page.extract_words() word sequence was influenced by the Y position of the text(for above case ,DAY seems to be in the same line but it has slighting difference so it is not after 1, but page.extract_text() is robust), so how to make page.extract_words() sequence same as page.extract_text()
I am glad if you may help on this case!
have a nice day,sir

**Reply by rahxphoon** [reply]
@jsvine I will appreciate if you have any solution for this case, that will help me a lot
Thanks so much

**Reply by cobaltautomationdev** [reply]
have you ever try this ? 

 Changing use_text_flow to True will use the PDF's underlying flow of characters as a guide for ordering and segmenting the words,

**Reply by rahxphoon** [reply]
@cobaltautomationdev 
seems like it got worse after **Changing use_text_flow to True**
 {
    'bottom': 420.2881999999999,
    'direction': 1,
    'doctop': 410.43819999999994,
    'text': '2',
    'top': 410.43819999999994,
    'upright': True,
    'x0': 56.43830000000005,
    'x1': 61.914900000000046
  },
  {
    'bottom': 420.6381999999999,
    'direction': 1,
    'doctop': 410.78819999999996,
    'text': 'DEMURRAGE - TRANSHIPMENT',
    'top': 410.78819999999996,
    'upright': True,
    'x0': 69.80000000000004,
    'x1': 220.20665000000005
  },
  {
    'bottom': 433.3381999999999,
    'direction': 1,
    'doctop': 423.48819999999995,
    'text': 'LOWBED',
    'top': 423.48819999999995,
    'upright': True,
    'x0': 69.80000000000005,
    'x1': 112.48990000000005
  },
  {
    'bottom': 443.76719999999995,
    'direction': 1,
    'doctop': 435.6672,
    'text': '(02/Dec/22)',
    'top': 435.6672,
    'upright': True,
    'x0': 92.70000000000005,
    'x1': 135.01440000000005
  },
  {
    'bottom': 453.96719999999993,
    'direction': 1,
    'doctop': 445.86719999999997,
    'text': '(TRLU6423563)',
    'top': 445.86719999999997,
    'upright': True,
    'x0': 94.95180000000005,
    'x1': 153.02070000000006
  },
  {
    'bottom': 470.18819999999994,
    'direction': 1,
    'doctop': 460.3382,
    'text': '100.00 DAY Z',
    'top': 460.3382,
    'upright': True,
    'x0': 411.9383,
    'x1': 560.7183500000001
  },
  {
    'bottom': 470.68819999999994,
    'direction': 1,
    'doctop': 460.8382,
    'text': '1 100.00',
    'top': 460.8382,
    'upright': True,
    'x0': 319.6883,
    'x1': 551.1596000000001
  },
  {
    'bottom': 472.37899999999996,
    'direction': 1,
    'doctop': 461.62899999999996,
    'text': '$',
    'top': 461.62899999999996,
    'upright': True,
    'x0': 477.95000000000005,
    'x1': 483.927
  }
  
  the extracted text is much more in disorder now, it extract the second(third) line before completing the first line...
  
![image](https://github.com/jsvine/pdfplumber/assets/16117379/7a6249ab-2a89-49d9-b70c-8ccf81b946f1)

**Reply by rahxphoon** [reply]
I think extract_text uses the PDF's underlying flow of characters as a guide for ordering and segmenting the words
I wish the sequence can also applied to extract_words (text field)

**Reply by cobaltautomationdev** [reply]
maybe you can try to extract this part as table? extract_table treat a line as a row.

**Reply by rahxphoon** [reply]
thanks, that will be considered in the future
may be I will try manual re-construct the extract_words() text sequence same as extract_text
since extract_text sequence is robust, but in the same time, I need the position from extract_words

**Reply by jsvine** [reply]
Hi @rahxphoon, can you provide a copy of the PDF?

In the meantime, you may have luck adjusting the `y_tolerance` parameter.
